pallet/src/lib.rs

```diff
@@ -710,6 +781,55 @@ pub mod pallet {
                 Event::FetchedGroupPermissions
             )
         }
+
+        /// The following extrinsics are used to delete entities
+        /// This deletes the role and refunds its deposit
+        /// This however, doesnt delete assignments under that role or refund their deposits
+        #[pallet::call_index(29)]
+        #[pallet::weight(T::WeightInfo::delete_role())]
```

Comment:
I think we should add these two extrinsic into the precompiled, but not in this MR. Let me create another Asana task